### ISSUE TITLE
Simplified factory method for `Constraint` interface

### ISSUE THREAD

**Issue by jenetics** [jenetics/jenetics#543]
It should be possible to create `Constraint` interfaces which work on the encoded type, instead of the `Phenotype` directly.
```java
public static <T, G extends Gene<?, G>, C extends Comparable<? super C>>
Constraint<G, C> of(
    final Encodec<T, G> encodec,
    final Predicate<? super T> validator,
    final Function<? super T, ? extends T> repairer
);
```
This depends on #542.

**Comment by jenetics** [comment]
Merged into r5.2.0 branch.
